utilities/copyoffload_migration.py

```
@@ -887,6 +888,186 @@ def _get_populate_pod_logs(
+        plan (Plan): The Plan CR resource with completed migration status.
+        expected_xcopy_used (int): Expected xcopyUsed annotation value (1 or 0).
+
+    Raises:
+        ValueError: If plan has no DiskTransfer tasks or annotations are missing.
+        AssertionError: If any task's xcopyUsed annotation doesn't match expected.
+    """
+    tasks = _get_disk_transfer_tasks(plan=plan)
+    expected_str = str(expected_xcopy_used)
+
+    LOGGER.info(
+        f"Verifying xcopyUsed Plan CR annotations: expecting '{expected_str}' on {len(tasks)} DiskTransfer task(s)"
```

Ruff (0.16.1)

[warning] Logging statement uses f-string

(G004)